tests/foreman/api/test_repository.py

```diff
@@ -113,43 +199,179 @@ def test_positive_create_with_gpg(self):
     @run_only_on('sat')
     def test_positive_create_same_name_different_orgs(self):
-        """Create two repos with the same name in two organizations.
+        """Create two repos with the same name in two different organizations.
 
-        @Assert: The two repositories are successfully created and use the
-        given name.
+        @Assert: The two repositories are successfully created and have given
+        name.
 
         @Feature: Repository
         """
-        repo1 = entities.Repository().create()
+        repo1 = entities.Repository(product=self.product).create()
         repo2 = entities.Repository(name=repo1.name).create()
-        for repo in (repo1, repo2, repo1.read(), repo2.read()):
-            self.assertEqual(repo.name, repo1.name)
+        self.assertEqual(repo1.name, repo2.name)
 
     @tier1
     @run_only_on('sat')
-    def test_positive_delete(self):
-        """Create a repository with attributes ``attrs`` and delete it.
+    def test_negative_create_name(self):
+        """Attempt to create repository with invalid names only.
 
-        @Assert: The repository cannot be fetched after deletion.
+        @Assert: A repository is not created and error is raised.
 
         @Feature: Repository
         """
-        for attrs in _test_data():
-            with self.subTest(attrs):
-                repo = entities.Repository(
-                    product=self.product,
-                    **attrs
-                ).create()
-                repo.delete()
+        for name in invalid_values_list():
+            with self.subTest(name):
                 with self.assertRaises(HTTPError):
-                    repo.read()
+                    entities.Repository(name=name).create()
+
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_same_name(self):
+        """Attempt to create a repository providing a name of already existent
+        entity
+
+        @Assert: Second repository is not created
+
+        @Feature: Repository
+        """
+        name = gen_string('alphanumeric')
+        entities.Repository(product=self.product, name=name).create()
+        with self.assertRaises(HTTPError):
+            entities.Repository(product=self.product, name=name).create()
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_label(self):
+        """Attempt to create repository with invalid label.
+
+        @Assert: A repository is not created and error is raised.
+
+        @Feature: Repository
+        """
+        with self.assertRaises(HTTPError):
+            entities.Repository(label=gen_string('utf8')).create()
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_url(self):
+        """Attempt to create repository with invalid url.
+
+        @Assert: A repository is not created and error is raised.
+
+        @Feature: Repository
+        """
+        with self.assertRaises(HTTPError):
+            entities.Repository(url=gen_string('alpha')).create()
```

Comment:
Maybe we could test the URL field by trying to provide multiple types of strings (HTML, UTF, ..) as we usually do for generic text fields. - applicable to all text fields across all pages.
This is suggestion and can be applied later,
not a blocker